compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs

```diff
@@ -857,6 +860,22 @@ data DefException = DefException
   }
   deriving (Eq, Data, Generic, NFData, Show)
 
+data DefInterface = DefInterface
+  { intLocation :: !(Maybe SourceLoc)
+  , intName :: !TypeConName
+  , intChoices :: ![InterfaceChoice]
```

Comment:
I think a `NameMap` would be the better choice here, which would be also in line with the `Template` data type.

Reply:
good idea e95de5f0d0ed2d0525b72d784807b14776f4473e